### PR TITLE
DX: remove empty sets from RuleSet

### PR DESCRIPTION
**Pull request by kubawerlos**
As requested by @keradus

### CODE REVIEW

**Comment by keradus** [comment]
i had to request it sooo long ago ;) can't remember about this

File: tests/RuleSetTest.php [hunk 1]
```diff
@@ -276,6 +276,33 @@ public function testSetDefinitionsAreSorted($setDefinitionName)
...
+    public function testSetDefinitionIsNotEmpty($setDefinitionName)
+    {
+        $ruleSet = RuleSet::create();
+
+        $method = new \ReflectionMethod(
+            \PhpCsFixer\RuleSet::class,
+            'getSetDefinition'
+        );
+
+        $method->setAccessible(true);
```
Comment (keradus) [inline tests/RuleSetTest.php]: `AccessibleObject` perhaps? it's already used in this file

File: tests/RuleSetTest.php [hunk 2]
```diff
@@ -276,6 +266,21 @@ public function testSetDefinitionsAreSorted($setDefinitionName)
...
     }
 
+    /**
+     * @dataProvider provideSetDefinitionNameCases
+     *
+     * @param string $setDefinitionName
+     */
+    public function testSetDefinitionIsNotEmpty($setDefinitionName)
+    {
+        $reflection = new AccessibleObject(RuleSet::create());
```
Comment (keradus) [inline tests/RuleSetTest.php]: `new RuleSet()` ?

Comment (kubawerlos) [reply tests/RuleSetTest.php]: Sure, done, any particular reason to change it here? I simply wonder why we even have this static constructor as it makes no difference comparing to normal creation.

Comment (keradus) [reply tests/RuleSetTest.php]: leftover (kept for bc reasons) of old php versions, where you could do `Foo::create()->bar();`, but you couldn't do it when using `new`

File: .php_cs.dist [hunk 3]
```diff
@@ -19,7 +19,7 @@ $finder = PhpCsFixer\Finder::create()
 $config = PhpCsFixer\Config::create()
     ->setRiskyAllowed(true)
     ->setRules([
-        '@PHP56Migration' => true,
```
Comment (keradus) [inline .php_cs.dist]: please mention that ruleset was removed in UPGRADE-v3.md

File: .php_cs.dist [hunk 4]
```diff
@@ -19,7 +19,7 @@ $finder = PhpCsFixer\Finder::create()
...
+        '@PHP56Migration:risky' => true,
```
Comment (keradus) [inline .php_cs.dist]: we don't need it here, it's part of `@PHPUnit60Migration:risky` that we are using already

Comment (kubawerlos) [reply .php_cs.dist]: Is it? I cannot see [it](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/src/RuleSet.php#L358).

Comment (keradus) [reply .php_cs.dist]: It is.

Comment (kubawerlos) [reply .php_cs.dist]: How?
I see `PHP56Migration:risky` is part of `@PHP70Migration:risky`.
Then `@PHP70Migration:risky` is part of `@PHP71Migration:risky`.
And that's it, how it becomes a part of `@PHPUnit60Migration:risky`?

Comment (kubawerlos) [reply .php_cs.dist]: @keradus take a look at: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4432/commits/7b65acacc2d589fcafe48e86e6310a8491b3fd63 - I have removed it and now it's OK to have `pow` function (which changing to `**` is part of `@PHP56Migration:risky` set).

Comment (SpacePossum) [reply .php_cs.dist]: `PHPUnit60Migration:risky` has `@PHPUnit56Migration:risky` it does indeed not have `PHP56Migration:risky` IIUC

Comment (keradus) [reply .php_cs.dist]: Aaah. Indeed, I read them both as phpunit rulesets. My bad, sorry

Comment (SpacePossum) [reply .php_cs.dist]: no worries :)

**Comment by SpacePossum** [comment]
Thank you @kubawerlos.

**Pull request merged**